Added capture options to allow for all capturable fields to be utilized

hatdragon opened the pull request:
This update will give users the ability to pass a StripeChargeCaptureOptions object into a capture request.  This will allow for additional API properties to be utilized for items such as destination[amount] and  statement_descriptor.

anelder-stripe commented on src/Stripe.Tests.XUnit/Stripe.Tests.XUnit.csproj:
```diff
@@ -27,4 +27,8 @@
     <ProjectReference Include="..\Stripe.net\Stripe.net.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
```
Can we nix this from the PR?

hatdragon replied:
I can, but VS is just going to add it back in.   Per Microsoft, it's intended behavior.  https://connect.microsoft.com/VisualStudio/feedback/details/800245/vs2013rc-adds-to-vs2012-c-project-section-itemgroup  

>This behavior is intentional. 
>
>To support third-party test frameworks, like NUnit and XUnit, Visual Studio 2012 
>loaded Test Explorer on solution open, regardless of whether it contained test 
>projects. This added seconds of delay to startup and solution open scenarios for all 
>users, majority of whom don't use tests.
>
>In Visual Studio 2013, we changed it so that Test Explorer package is loaded only 
>when the solution contains one or more test projects. Test projects are identified in 
>two different ways. Projects created from one of the built-in unit test project 
>templates are identified using project type GUIDs. Other types of projects, such as 
>Class Library project with XUnit or NUnit tests, are identified by Test Explorer 
>during first test discovery and “tagged” with the <Service/> item. 
>
>You can avoid Visual Studio automatically adding the <Service/> item to your test 
>projects by creating them from any of the built-in unit test project templates. In 
>other words, select the "Unit Test Project" template instead of the "Class Library" 
>template in the New Project dialog.

anelder-stripe commented on src/Stripe.net.Tests/Stripe.net.Tests.csproj:
```diff
@@ -49,4 +49,8 @@
     <Reference Include="System" />
   </ItemGroup>
 
+  <ItemGroup>
```
This too please!  :)

anelder-stripe commented on src/Stripe.net.Tests/charges/test_data/stripe_charge_capture_options.cs:
```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections.Generic;
+
+namespace Stripe.Tests.test_data
```
Would you be willing to either (a) move these tests to the XUnit project or (b) write some more tests for the XUnit project to include here?  We're going to be removing the mspec tests.  :)

hatdragon replied:
Sure, I'll take a look tomorrow

hatdragon commented:
Added additional xunit tests and cleaned up a few small things that I missed with the options in my original testing.   Please take a look and let me know if there are any additional things we need to wrap this one up.

hatdragon commented:
@anelder-stripe?

remi-stripe commented:
r? @brandur-stripe @fred-stripe 
Bumping this one to make sure it gets handled soon

remi-stripe commented:
@hatdragon I totally missed that you did this a few months ago and I did my own version in https://github.com/stripe/stripe-dotnet/pull/1019. It has been merged though so you should be all set.
